File: notification-service/src/types/LikeNotification.ts
import { NotificationPayload } from "./NotificationPayload";
import { Notification } from "./Notification";

export class LikeNotification extends Notification {
  build(): NotificationPayload {
    return {
      to: this.record.destinationToken,
      title: "Your post got a like!",
      body: `${this.record.likedBy} liked your spot.`,
      data: { postId: this.record.postId, type: "LIKE" },
    };
  }
}

File: posts-service/src/main/java/org/example/postsservice/models/likes/LikeNotification.java
package org.example.postsservice.models.likes;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class LikeNotification {
    private final String notificationType = "LIKE";
    private Long postId;
    private String likedBy;
    private String destinationToken;
}
